static/app/components/inputGroup.tsx

```diff
@@ -0,0 +1,211 @@
+import {
+  createContext,
+  forwardRef,
+  useContext,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import styled from '@emotion/styled';
+
+import _TextArea from 'sentry/components/forms/controls/textarea';
+import _Input, {InputProps} from 'sentry/components/input';
+import space from 'sentry/styles/space';
+import {FormSize} from 'sentry/utils/theme';
+
+interface InputContext {
+  /**
+   * Props to be passed to `Input`. When `InputGroup` is used, it is
+   * recommended to add input props (`size`, `disabled`) to `InputGroup`
+   * rather than `Input`, so that other elements in the group
+   * (`InputLeadingItems`, `InputTrailingItems`) also know about them.
+   */
+  inputProps: Partial<InputProps>;
```

Comment:
Damn, I don't particularly like that the API for the input props is now split between `Input` and `InputGroup`.

I think it might be confusing what props to use where. Should we maybe force all UI related input props onto InputGroup and disallow them on the exported Input